cinema/serializers.py

```diff
@@ -68,13 +84,53 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SerializerMethodField(method_name="get_tickets")
+
+    def get_tickets(self, obj):
+        return [
+            {"row": ticket.row, "seat": ticket.seat}
+            for ticket in obj.tickets.all()
+        ]
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat", "movie_session")
+
+
+class TicketListSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
+
+    class Meta:
+        model = Order
+        fields = ("id", "tickets", "created_at")
+
+    def create(self, validated_data):
+        with transaction.atomic():
+            tickets_data = validated_data.pop("tickets")
+            order = Order.objects.create(**validated_data)
+            for ticket_data in tickets_data:
+                Ticket.objects.create(order=order, **ticket_data)
+            return order
```

Comment:
you can use simple decorator here, since whole func is a transaction.
and an advice:
Using `bulk_create` can significantly improve efficiency when creating multiple objects, like `Ticket` instances. Consider restructuring the code to utilize `bulk_create` for creating `Ticket` objects instead of iterating through them in a loop.
For example:
```
ticket_objects = [Ticket(order=order, **ticket_data) for ticket_data in tickets]
Ticket.objects.bulk_create(ticket_objects)
```
using list comprehensions for objects defines best arrangement of memory inside python code, and bulk_create is more efficient, than going in for loop for an object.